SessionMessagingKit/Threads/TSThread.m

```diff
@@ -261,56 +261,39 @@ - (NSUInteger)numberOfInteractions
 
 - (NSUInteger)unreadMessageCountWithTransaction:(YapDatabaseReadTransaction *)transaction
 {
-    __block NSUInteger count = 0;
-
     YapDatabaseViewTransaction *unreadMessages = [transaction ext:TSUnreadDatabaseViewExtensionName];
-    [unreadMessages enumerateKeysAndObjectsInGroup:self.uniqueId
-                                        usingBlock:^(NSString *collection, NSString *key, id object, NSUInteger index, BOOL *stop) {
-        if (![object conformsToProtocol:@protocol(OWSReadTracking)]) {
-            return;
-        }
-        id<OWSReadTracking> unread = (id<OWSReadTracking>)object;
-        if (unread.read) {
-            NSLog(@"Found an already read message in the * unread * messages list.");
-            return;
-        }
-        count += 1;
-    }];
-
-    return count;
-}
-
-- (BOOL)hasUnreadMentionMessageWithTransaction:(YapDatabaseReadTransaction *)transaction
-{
-    __block BOOL hasUnreadMention = false;
+    return [unreadMessages numberOfItemsInGroup:self.uniqueId];
     
-    YapDatabaseViewTransaction *unreadMessages = [transaction ext:TSUnreadDatabaseViewExtensionName];
-    [unreadMessages enumerateKeysAndObjectsInGroup:self.uniqueId
-                                        usingBlock:^(NSString *collection, NSString *key, id object, NSUInteger index, BOOL *stop) {
-        if (![object isKindOfClass:[TSIncomingMessage class]]) {
-            return;
-        }
-        TSIncomingMessage* unreadMessage = (TSIncomingMessage*)object;
-        if (unreadMessage.read) {
-            NSLog(@"Found an already read message in the * unread * messages list.");
-            return;
-        }
-        
-        if (unreadMessage.isUserMentioned) {
-            hasUnreadMention = true;
-            *stop = YES;
-        }
-    }];
-
-    return hasUnreadMention;
     
+    // FIXME: Why did we have to do as the following?
```

Comment:
Reads to me like it was a sanity check (ie. to flag when it found a read message in the unread messages group) - I think it's safe to delete it (if we start hearing about buggy counts then we can use git history to look at the old code)

Reply:
OK, I'll leave the old code there for now and see if there is any bug reports raised after we release this PR. If everything's good we can clean it up later.

Reply:
No worries